lib/did_you_mean/finders/name_error_finders/similar_class_finder.rb

```diff
@@ -32,6 +33,13 @@ def scopes
 
     private
 
+    def autoload_class_name_inflector
+      return unless defined?(Rails)
+      name_is_singular = name_from_message.to_s.pluralize.singularize == name_from_message.to_s
+      class_name_inflector = name_is_singular ? name_from_message.to_s.pluralize : name_from_message.to_s.singularize
+      class_name_inflector.safe_constantize
+    end
+
```

Comment:
If I'm in a Rails console in development mode and type Subscribers.first, then I get "undefined constant" with no suggestion.  I was confused at first as to whydid_you_mean did not suggest the correct Subscriber.first.  However, if I type Subscriber (to load the class) and then type Subscribers.first, I then get the suggestion.

I'm not sure I see the harm in auto-loading a single class when an error occurs to make a single suggestion.  It's only executed for the constant they they mistyped (not all the possible constants available).  And it's only executed when an error occurs and did_you_mean is making a suggestion.